core/pkg/hostsensorutils/hostsensorgetfrompod.go

```diff
@@ -323,18 +323,9 @@ func (hsh *HostSensorHandler) CollectResources(ctx context.Context) ([]hostsenso
 		res = append(res, kcData...)
 	}
 
-	// GetControlPlaneInfo
-	kcData, err = hsh.GetControlPlaneInfo(ctx)
-	if err != nil {
-		addInfoToMap(ControlPlaneInfo, infoMap, err)
-		logger.L().Ctx(ctx).Warning(err.Error())
-	}
-	if len(kcData) > 0 {
-		res = append(res, kcData...)
-	}
-
 	// GetCloudProviderInfo
 	kcData, err = hsh.GetCloudProviderInfo(ctx)
+	isCloudProvider := (kcData != nil)
```

Comment:
[Question]: is kcData nil in cases where we are not running on cloud provider?

Reply:
I supposed it was like so, but I was wrong.
I'm going to provide a fix for this asap.